DictionaryCompressor/CommandLineParsing/Parser/CommandLineParser.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.IO;
+using System.Linq;
+
+namespace DictionaryCompressor.CommandLineParsing.Parser
+{
+    public sealed class CommandLineParser : ICommandLineParser
+    {
+        private readonly IFileCompressor _fileCompressor;
+        private Command _rootCommand;
+
+        public CommandLineParser(IFileCompressor fileCompressor)
+        {
+            _fileCompressor = fileCompressor;
+        }
+
+        public void Parse(string[] args)
+        {
+            _rootCommand = new RootCommand
+            {
+                new Option(
+                    "--files",
+                    "Files location")
+                {
+                    Argument = new Argument<IEnumerable<string>>(),
+                },
+                new Option(
+                    "--directories",
+                    "Directories Location")
+                {
+                    Argument = new Argument<IEnumerable<string>>(),
+                },
+                new Option(
+                    "--output",
+                    "Output path")
+                {
+                    Argument = new Argument<string>(),
+                },
+            };
+            IEnumerable<FileSystemInfo> paths = new List<FileSystemInfo>();
+            _rootCommand.Handler = CommandHandler.Create<List<string>, List<string>, string>((files, directories, output) =>
+                {
```

Comment:
I think it would be user friendly if you printed a message if the user forgot to used neither `--files` nor `--directorie`. Is it possible to return the `--help` output somehow?